extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/MethodNameParser.java

```diff
@@ -97,18 +97,24 @@ public Result parse(MethodInfo methodInfo) {
         int firstIndex = methodName.indexOf("First");
         int topIndex = methodName.indexOf("Top");
         if ((firstIndex != -1) || (topIndex != -1)) {
-            try {
-                String topCountStr = methodName.substring(Math.max(firstIndex, topIndex), byIndex)
-                        .replace("Top", "").replace("First", "");
-                if (topCountStr.isEmpty()) {
-                    topCount = 1;
-                } else {
-                    topCount = Integer.valueOf(topCountStr);
+            int effectiveFirstOrTopIndex = (topIndex > -1 && firstIndex > -1) ? Math.max(topIndex, firstIndex)
+                    : Math.max(firstIndex, topIndex);
```

Comment:
How is `Math.max(topIndex, firstIndex)` significantly different from `Math.max(firstIndex, topIndex)`?

This will still favor a "First" or "Top" in the back of the name instead of taking the very first one.
IMO `min()` is the right way, but with filtering out -1.